source/extensions/load_balancing_policies/override_host/override_host_filter_state.h

```diff
@@ -16,15 +17,22 @@ class OverrideHostFilterState : public StreamInfo::FilterState::Object {
 public:
   static constexpr absl::string_view kFilterStateKey =
       "envoy.extensions.load_balancing_policies.override_host.filter_state";
-  OverrideHostFilterState() = default;
 
-  uint64_t fallbackHostIndex() const { return fallback_host_index_; }
-  void setFallbackHostIndex(uint64_t fallback_host_index) {
-    fallback_host_index_ = fallback_host_index;
+  OverrideHostFilterState(std::string&& host_list) : host_list_(std::move(host_list)) {
+    for (absl::string_view host : absl::StrSplit(host_list_, ',', absl::SkipWhitespace())) {
+      selected_hosts_.push_back(absl::StripAsciiWhitespace(host));
+    }
   }
 
+  uint64_t hostIndex() const { return host_index_; }
+  void setHostIndex(uint64_t host_index) { host_index_ = host_index; }
+
+  absl::Span<const absl::string_view> selectedHosts() const { return selected_hosts_; }
+
 private:
-  uint64_t fallback_host_index_ = 1;
+  const std::string host_list_;
```

Comment:
Is this used for anything?

Reply:
This host list will take the ownership of the addresses list. And the vector only store the string_view. This ensure we only need once heap allocation in most cases.

Reply:
In practice this optimization will never be needed as the number of hosts is very small. However there is more complexity. This stores references to class members, so the class needs at the very least to be non copyable and non movable.

However storing references to class members is generally considered an antipattern, unless absolutely needed. There is really no evidence that this complexity is justified in this case. You certainly need to document what is happening here as it is not obvious.

Reply:
Maybe I am too sensitive to the possible frequent heap allocations because the https://github.com/envoyproxy/envoy/issues/19103

But the FilterState object will never be copied or moved and is managed by shared ptr alwaysly. Let me re-thinking it....

Reply:
I did a lazy parsing way. Because in our implemention, we always will keep a index as start of next start. Then we can use the index to record the char offset of the host we have consumed.
And at next time, we can get host by finding ',' start from the offset.

This bring additional advantage is we needn't to maintain the index/offset by the caller. The index/offset will be maintained by OverrideHostFilterState internally.